mjattbot.js

```diff
@@ -58,33 +60,42 @@ function pokeCatch() {
 }
 
 client.on("message", message => {
-  if (message.author.bot) return;
-  if (message.channel.type === "dm") {
-    message.channel.send(
-      "Please try this in a public guild. Error: DM channel"
-    );
-    return;
-  }
-  if (!message.content.startsWith(key.prefix)) return;
+  if (message.author.bot) return; // ignore bots to avoid infinite loop
+  if (message.channel.type === "dm") return message.channel.send("Please try this in a public guild. Error: DM channel"); // cleanup, code looks bit cleaner here imo
+  
+  if (!message.content.startsWith(key.prefix)) return; // ignore msgs without prefix
 
   let command = message.content.split(" ")[0];
   command = command.slice(key.prefix.length);
+  // console.log(command);
 
   let args = message.content.split(" ").slice(1);
 
   try {
     let cmd = require(`./commands/${command}.js`);
-    cmd.run(client, message, args);
     count++;
     console.log(
-      `Message Number: ${count} Command: ${command} Args: ${args.join(" ")} Author: ${message.author.username}`
+      `Message Number: ${count} Command: ${command} Args: ${args.join(" ") ? args.join(" ") : "no arguments"} Author: ${message.author.username}`
     );
+    cmd.run(client, message, args);
   } catch (err) {
-    console.log(err);
+    try {
```

Comment:
Rather than waiting for the command to error, might make more sense to check if the file exists. `fs.existsSync()` would work well, or the async version with a promise/callback would work too. As theoretically a command could error for another reason not just "couldn't find it".